projects/optic/src/commands/bundle/bundle.ts

```diff
@@ -0,0 +1,507 @@
+      strict: false,
+      denormalize: false,
+    });
+  } catch (e) {
+    console.error(e instanceof Error ? e.message : e);
+    throw new UserError();
+  }
+};
+
+const bundleAction =
+  (config: OpticCliConfig) => async (filePath: string | undefined, options) => {
+    const { o, filterXExtensions, includeXExtensions } = options;
+
+    const filterExtensions = (filterXExtensions || '')
+      .split(/[ ,]+/)
+      .filter((extension) => extension.startsWith('x-'));
+    const includeExtensions = (includeXExtensions || '')
+      .split(/[ ,]+/)
+      .filter((extension) => extension.startsWith('x-'));
+
+    let parsedFile: ParseResult;
+    if (filePath) {
+      parsedFile = await getSpec(filePath, config);
+
+      const updatedSpec = bundle(parsedFile.jsonLike, parsedFile.sourcemap);
+
+      if (includeExtensions.length) {
+        Object.entries(updatedSpec.paths).forEach(([path, operations]) => {
+          Object.entries(operations!).forEach(([key, operation]) => {
+            if (key === 'parameters') return;
+            if (
+              operation &&
+              !includeExtensions.some((extension) =>
+                Boolean(operation[extension])
+              )
+            ) {
+              // @ts-ignore
+              delete updatedSpec.paths![path]![key]!;
+              const otherKeys = Object.keys(updatedSpec.paths![path] || {});
+              if (
+                otherKeys.length === 0 ||
+                (otherKeys.length === 1 && otherKeys[0] === 'parameters')
+              ) {
+                delete updatedSpec.paths![path];
+              }
+            }
+          });
+        });
+      }
+
+      if (filterExtensions.length) {
+        Object.entries(updatedSpec.paths).forEach(([path, operations]) => {
+          Object.entries(operations!).forEach(([key, operation]) => {
+            if (key === 'parameters') return;
+            // should filter
+            if (
+              operation &&
+              filterExtensions.some((extension) =>
+                Boolean(operation[extension])
+              )
+            ) {
+              // @ts-ignore
+              delete updatedSpec.paths![path]![key]!;
+              const otherKeys = Object.keys(updatedSpec.paths![path] || {});
+              if (
+                otherKeys.length === 0 ||
+                (otherKeys.length === 1 && otherKeys[0] === 'parameters')
+              ) {
+                delete updatedSpec.paths![path];
+              }
+            }
+          });
+        });
+      }
+
+      const yamlOut = () =>
+        yaml.stringify(updatedSpec, {
+          defaultStringType: 'QUOTE_DOUBLE',
+        });
+
+      if (o) {
+        // write to file
+        const outputPath = path.resolve(o);
+        await fs.writeFile(
+          outputPath,
+          isYaml(o) ? yamlOut() : JSON.stringify(updatedSpec, null, 2)
+        );
+        console.log('wrote bundled spec to ' + path.resolve(o));
```

Comment:
should use `../logger.ts` instead of console.logs so we can control the log levels